src/engine/sparqlExpressions/RelationalExpressions.cpp

```diff
@@ -409,6 +409,55 @@ ExpressionResult InExpression::evaluate(
   return result;
 }
 
+// _____________________________________________________________________________
+template <Comparison comp>
+std::optional<std::vector<PrefilterExprVariablePair>>
+RelationalExpression<comp>::getPrefilterExpressionForMetadata(
+    [[maybe_unused]] bool isNegated) const {
+  AD_CORRECTNESS_CHECK(children_.size() == 2);
+  const SparqlExpression::Ptr& child0 = children_.at(0);
+  const SparqlExpression::Ptr& child1 = children_.at(1);
+
+  const auto checkBinarySearchEvaluable =
+      [](const SparqlExpression::Ptr& child0,
+         const SparqlExpression::Ptr& child1)
+      -> std::optional<std::pair<ValueId, Variable>> {
+    if (const auto* literalValueId =
+            dynamic_cast<const IdExpression*>(child0.get())) {
+      if (const auto* literalVariable =
+              dynamic_cast<const VariableExpression*>(child1.get())) {
+        return std::make_pair(literalValueId->value(),
+                              literalVariable->value());
+      }
+    }
+    return std::nullopt;
+  };
+
+  const auto createPrefilterExprVariablePair =
+      [](const std::pair<ValueId, Variable>& valuePair)
+      -> std::vector<PrefilterExprVariablePair> {
+    std::vector<PrefilterExprVariablePair> pairVec;
+    pairVec.emplace_back(
+        std::make_unique<prefilterExpressions::RelationalExpression<comp>>(
+            valuePair.first),
+        valuePair.second);
+    return pairVec;
+  };
+  // Option 1: child0 is a (constant) reference ValueId, while child1 contains a
+  // Variable to the respective column on which we want to filter.
+  const auto& optionalPair1 = checkBinarySearchEvaluable(child0, child1);
+  if (optionalPair1.has_value()) {
+    return createPrefilterExprVariablePair(optionalPair1.value());
+  }
+  // Option 2: child1 is a (constant) reference ValueId, while child0 contains a
+  // Variable to the respective column on which we want to filter.
+  const auto& optionalPair2 = checkBinarySearchEvaluable(child1, child0);
+  if (optionalPair2.has_value()) {
+    return createPrefilterExprVariablePair(optionalPair2.value());
+  }
```

Comment:
I think this is wrong.
If you switch the pair (which you do here), I think you also need to switch the expression.
E.g. if your expression here is ` 5 > x` then you currently do `PrefilterGreater(5), x` which is wrong, you have to get the complement here. You need some logic to get the "reversed" expression.
Please also add a unit test that would have caught this bug.

Reply:
Yes indeed, thanks. Fixed it, with help of the already existing `logicalComplement` from  `PrefilterExpressions`

Reply:
Yes indeed, thanks. Those test cases have been added to test `getPrefilterExpressionFromSparqlRelational`